scripts/devnet-curio/docker-compose.yml

```diff
@@ -0,0 +1,363 @@
+      - FOREST_WAFFLE_HEIGHT=${WAFFLE_HEIGHT}
+    networks:
+      - devnet
+    ports:
+      - ${FOREST_RPC_PORT}:${FOREST_RPC_PORT}
+    entrypoint: [ "/bin/bash", "-c" ]
+    command:
+      - |
+        set -euxo pipefail
+        if [ ! -f ${FOREST_DATA_DIR}/forest_config.toml ]; then
+          export NETWORK_NAME=$$(grep -o \"localnet.*\" ${LOTUS_DATA_DIR}/localnet.json)
+          cp /forest/forest_config.toml.tpl ${FOREST_DATA_DIR}/forest_config.toml
+          echo name = $$NETWORK_NAME >> ${FOREST_DATA_DIR}/forest_config.toml
+        fi
+        forest --genesis ${LOTUS_DATA_DIR}/devgen.car --config ${FOREST_DATA_DIR}/forest_config.toml --save-token ${FOREST_DATA_DIR}/token.jwt --rpc-address 0.0.0.0:${FOREST_RPC_PORT}
+
+  # At the moment of writing, Forest was not able to connect to a devnet node using its config.
+  # This is a workaround to force the connection.
+  forest_connecter:
+    depends_on:
+      forest:
+        condition: service_healthy
+    build:
+      context: ../../.
+      dockerfile: ${FOREST_DOCKERFILE_OVERRIDE:-Dockerfile}
+    volumes:
+      - ./data/lotus:${LOTUS_DATA_DIR}
+      - ./data/forest:${FOREST_DATA_DIR}
+    networks:
+      - devnet
+    entrypoint: [ "/bin/bash", "-c" ]
+    command:
+      - |
+        set -euxo pipefail
+        export TOKEN=$$(cat ${FOREST_DATA_DIR}/token.jwt)
+        export FULLNODE_API_INFO=$$TOKEN:/dns/forest/tcp/${FOREST_RPC_PORT}/http
+        forest-wallet --remote-wallet import ${LOTUS_DATA_DIR}/genesis-sectors/pre-seal-${MINER_ACTOR_ADDRESS}.key || true
+        forest-cli net connect /dns/lotus/tcp/${LOTUS_P2P_PORT}/p2p/$$(cat ${LOTUS_DATA_DIR}/PEER_ID)
+  f3:
+    user: root
+    depends_on:
+      forest_connecter:
+        condition: service_completed_successfully
+    build:
+      context: ../../f3-sidecar
+      dockerfile: Dockerfile
+    container_name: f3
+    volumes:
+      - ./data/lotus:${LOTUS_DATA_DIR}
+      - ./data/forest:${FOREST_DATA_DIR}
+    ports:
+      - ${F3_RPC_PORT}:${F3_RPC_PORT}
+    networks:
+      - devnet
+    entrypoint: [ "/bin/bash", "-c" ]
+    command:
+      - |
+        set -euxo pipefail
+        f3-sidecar -finality ${F3_FINALITY} -rpc http://forest:${FOREST_RPC_PORT}/rpc/v1 -f3-rpc 0.0.0.0:${F3_RPC_PORT}
+
+  curio:
+    depends_on:
+      forest:
+        condition: service_healthy
+      yugabyte:
+        condition: service_started
+      lotus_node:
+        condition: service_healthy
+    container_name: curio
+    image: ${CURIO_IMAGE}
+    init: true
+    ports:
+      - "12300:12300" # API
+      - "4701:4701" # UI
+      - "32100:32100" # Market
+    restart: unless-stopped
+    volumes:
+      - ./data/curio:/var/lib/curio:rw
+      - filecoin-proofs:/var/tmp/filecoin-proof-parameters:rw
+      - ./data/lotus:${LOTUS_DATA_DIR}
+      - ./data/forest:${FOREST_DATA_DIR}
+    env_file:
+      - curio.env
+    networks:
+      - devnet
+    entrypoint: [ "/bin/bash", "-c" ]
+    command:
+      - |
```

Comment:
nit: this command here seems way too long and convoluted to be embedded into `docker-compose.yaml`. However I also understand that it might be nice to be able to read the logic directly as opposed to navigating to the corresponding bash script. Debugging is definitely easier via a separate script though.

I leave it up to you, I'd likely have gone with a dedicated `.sh` file.

Reply:
Why is debugging easier with a separate script?

What's a good way to run a script here?

Reply:
If I had to debug this - I'd mount this script and let the entrypoint be just some unrelated long-running process to keep the container going. Then I would `docker ssh` and debug the script by running and modifying it as needed. 
Normally this should be faster than having to restart the container or the whole compose setup.

Reply:
I agree that it would be beneficial to move it into a script.

1. Compose parts should ideally live on a similar abstraction level. It should be possible to take a look and assess what different containers are doing. This is not possible with a lengthy script exposing internal logic.
2. Shell scripts inside compose files are not linted by the shellcheck.

Reply:
How do you move it to a script? I also asked Roman and didn't get an answer. You Docker gods have to take pity on a Docker peasant like me. I failed to Google an answer, and I see no examples in our docker-compose files.

Reply:
We do it in the sync check. https://github.com/ChainSafe/forest-iac/blob/093f9965ad0e148f3b5cc558e78720e9df3228e8/tf-managed/modules/sync-check/service/docker-compose.yml#L91-L97

Please let me know if you need any help with that.

Reply:
Done.

Reply:
Oh look, linter errors! No green checkmark, no review!!!

Reply:
> Why is debugging easier with a separate script?
> 
> What's a good way to run a script here?

I didn't understand the question, excuses. You just make sure the script is in the filesystem and then run it :)